packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart

```diff
@@ -132,33 +129,36 @@ class DebuggerController extends DisposableController
 
   final bool initialSwitchToIsolate;
 
-  IsolateState get isolateDebuggerState =>
+  IsolateState? get isolateDebuggerState =>
       serviceManager.isolateManager.isolateDebuggerState(isolateRef);
 
-  VmServiceWrapper get _service => serviceManager.service;
+  VmServiceWrapper get _service {
+    assert(serviceManager.hasService);
+    return serviceManager.service!;
+  }
 
   /// Cache of autocomplete matches to show for a library when that library is
   /// imported.
   ///
   /// This cache includes autocompletes from libraries exported by the library
   /// but does not include autocompletes for libraries imported by this library.
 
-  Map<LibraryRef, Future<Set<String>>> libraryMemberAutocompleteCache = {};
+  Map<LibraryRef?, Future<Set<String?>>> libraryMemberAutocompleteCache = {};
 
   /// Cache of autocomplete matches for a library for code written within that
   /// library.
   ///
   /// This cache includes autocompletes from all libraries imported and exported
   /// by the library as well as all private autocompletes for the library.
-  Map<LibraryRef, Future<Set<String>>>
+  Map<LibraryRef, Future<Set<String?>>>
       libraryMemberAndImportsAutocompleteCache = {};
 
   ProgramExplorerController get programExplorerController =>
       _programExplorerController;
-  ProgramExplorerController _programExplorerController;
+  late final ProgramExplorerController _programExplorerController;
```

Comment:
This getter doesn't appear to be adding any value. Can we just make the late final var public?
`late final ProgramExplorerController programExplorerController;`